library/core/functions.general.php

```diff
@@ -3926,19 +3926,40 @@ function userAgentType($value = null) {
             $type = $value;
         }
 
+        if (defined('TESTMODE_ENABLED') && constant('TESTMODE_ENABLED')) {
+            $type = null;
+        }
```

Comment:
I'm not the biggest fan of doing explicit test checking in functions like this.

Since this function is just being called directly in the test could we use some parameter like `ignoreCache` or `skipCache` instead?

Reply:
I tried your suggestion and it looks a lot more sloppy IMO and then adds a feature to the weird function that we would then have to support. The choice is to either do significantly more refactoring or just us the constant which is consistent with the way some other dev who will remain nameless did it.

Reply:
Alright, I opted for **false** clearing the cache.